internal/extproc/chatcompletion_processor.go

```diff
@@ -84,6 +84,25 @@ func (c *chatCompletionProcessor) ProcessRequestBody(ctx context.Context, rawBod
 	c.requestHeaders[c.config.modelNameHeaderKey] = model
 	b, err := c.config.router.Calculate(c.requestHeaders)
 	if err != nil {
+		// if err.Error() == "no matching rule found" {
```

Comment:
Instead of doing this and looking the error tree like you do, let's create a proper `ErrRouteNotFound` in the `filterapi/x` package. This was all implementations of Calculate can return that error consistently.
Then you can modify the default Calculate method to return that error, and here can cleanly just use `errors.Is` and return the immediate response directly.